src/main.js

```diff
@@ -37,6 +37,12 @@ app.on('ready', () => {
       slashes: true
     })
   )
+
+  // Work-around for electron/chrome 51+ onbeforeunload behavior
+  // which prevents the app window to close if not invlidated.
+  mainWindow.webContents.once('dom-ready', () => {
+    mainWindow.webContents.executeJavaScript("window.onbeforeunload = null")
```

Comment:
nice catch - how did you figure out that obscure solution?

Reply:
A lot of head scratching 🤔 and comparing behaviors between electron and chrome to find where went wrong. 

I figured it is probably better to inject this fix from outside the remix distribution rather than modifying the code within since we are pulling in new remix releases frequently.

After confirming that nothing much is allowed to happen within onbeforeunload event

> onbeforeunload dialogs are used for two things on the Modern Web:
> 1. Preventing users from inadvertently losing data.
> 2. Scamming users.

I've decided to generalize this bug's solution to just invalidating that entire event handler, until electron updates and this workaround is no longer needed.

Reply:
@acrylix awesome man, nice work! I upgraded all other deps too, as well the URL paths to remix static files to ensure the appcache works for persisting solc compiler for offline usage as well. See https://github.com/horizon-games/remix-app/releases/tag/0.2.0 :)